salt/states/boto_ec2.py

```diff
@@ -1752,5 +1749,169 @@ def private_ips_absent(name, network_interface_name=None, network_interface_id=N
+
+
+def secgroup_absent(
+        name,
+        device_index=0,
+        secgroup_names=None,
+        region=None,
+        key=None,
+        keyid=None,
+        profile=None):
+    '''
+    Ensure security groups are absent for existing instance(s) with given
+    instance name (wildcards allowed).
+
+    name
+        Instance name (wildcards allowed)
+
+    device_index
+        Network Device Index (default 0)
+
+    secgroup_names
+        List of security group names (which must exist) to ensure absence on
+        the given instances
+
+    region
+        Region to connect to.
+
+    key
+        Secret key to be used.
+
+    keyid
+        Access key to be used.
+
+    profile
+        A dict with region, key and keyid, or a pillar key (string)
+        that contains a dict with region, key and keyid.
+
+    .. code_block:: yaml
+        remove_security_groups:
+          boto_ec2.secgroup_absent:
+          - name: "cool_server-*"
+          - secgroup_names:
+            - "security_group_1"
+            - "security_group_2"
+    '''
+    return _secgroup_change(name, device_index, secgroup_names, operation='remove',
+                            region=region, key=key, keyid=keyid, profile=profile)
+
+
+def secgroup_equals(
+        name,
+        device_index=0,
+        secgroup_names=None,
+        region=None,
+        key=None,
+        keyid=None,
+        profile=None):
+    '''
+    Ensure security groups are present for existing instance(s) with given
+    instance name (wildcards allowed). All other security groups are removed
+    from the instance(s).
+
+    name
+        Instance name (wildcards allowed)
+
+    device_index
+        Network Device Index (default 0) to operate on.
+
+    secgroup_names
+        List of security group names (which must exist) to ensure absence on
+        the given instances
+
+    region
+        Region to connect to.
+
+    key
+        Secret key to be used.
+
+    keyid
+        Access key to be used.
+
+    profile
+        A dict with region, key and keyid, or a pillar key (string)
+        that contains a dict with region, key and keyid.
+
+    .. code_block:: yaml
+        set_security_groups:
+          boto_ec2.secgroup_equals:
+          - name: "cool_server-*"
+          - secgroup_names:
+            - "security_group_1"
+            - "security_group_2"
+    '''
+    return _secgroup_change(name, device_index, secgroup_names, operation='equals',
+                            region=region, key=key, keyid=keyid, profile=profile)
+
+
+def _secgroup_change(
+        name,
+        device_index=0,
+        secgroup_names=None,
+        operation=None,
+        region=None,
+        key=None,
+        keyid=None,
+        profile=None):
+    '''
+    Multifunction wrapper function for secgroup_present, secgroup_absent and secgroup_equals
+    '''
+    ret = {'name': name, 'result': True, 'changes': {}, 'comment': ''}
+    result = __salt__['boto_ec2.modify_instances_secgroup'](
```

Comment:
Ok, I'll move the code back to the state module, but that will cause some code duplication, since each state function will have to fetch the current state, determine the differences and then check for testmode and actualize the changes (or not).